sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -908,21 +908,35 @@ object ScalaJSPluginInternal {
       scalaJSModuleInitializers :=
         (scalaJSModuleInitializers in (This, Zero, This)).value,
 
-      scalaJSModuleInitializers ++= {
+      scalaJSModuleInitializers ++= Def.taskDyn {
         if (scalaJSUseMainModuleInitializer.value) {
-          Seq(scalaJSMainModuleInitializer.value.getOrElse {
-            throw new MessageOnlyException(
-                "No main module initializer was specified (possibly because " +
-                "no or multiple main classes were found), but " +
-                "scalaJSUseMainModuleInitializer was set to true. " +
-                "You can explicitly specify it either with " +
-                "`mainClass := Some(...)` or with " +
-                "`scalaJSMainModuleInitializer := Some(...)`")
-          })
+          Def.task {
+            Seq(scalaJSMainModuleInitializer.value.getOrElse {
+              scalaJSDiscoveredMainClasses.value.keys.toSeq match {
+                case Seq() =>
+                  throw new MessageOnlyException(
+                    """No main module initializer was specified, but
+                      | scalaJSUseMainModuleInitializer was set to true.
+                      | You can explicitly specify it either with
+                      | `mainClass := Some(...)` or with
+                      | `scalaJSMainModuleInitializer := Some(...)`"""
+                      .stripMargin)
+                case multiple =>
+                  throw new MessageOnlyException(
+                    s"""Multiple main classes (${multiple}) were found.
+                       | Main class should be specified exact once either with
```

Comment:
I'm not quite convinced by the English formulation here. It could be my English (I'm not a native speaker). Perhaps consider:
```suggestion
                       | You can explicitly specify the one you want with
```
(it's more in line with the other error message)

Reply:
Applied suggestion in https://github.com/scala-js/scala-js/pull/3724/commits/eaa336487bd07a55b9bd8e5ed9311373157581ff